WordPress/openverse-frontend
# Fix missing nuxt types

## Description


In the Vue SFC typescript PR #1163 I added the `types` compiler option to our tsconfig and this removed some critical nuxt types from global scope. Adding them back also revealed that I'd accidentally duplicated the work that the Nuxt vue-i18n module does to make the types work so I was able to remove some code as well.

I also added an ESLint rule to prevent accidentally destructuring the `i18n` object as it will result in a failure if you do and try to call one of the methods destructured due to the dependency on `this` in the method.

## Testing Instructions

Go to `use-i18n.ts` and call the function, make sure it has the correct types when you inspect the type information in your editor (try to use some of the properties like `.t` for example). This tests the removal of the code in the nuxt__types/index.d.ts module and also tests that the types from the module are being added.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
